NebulaHost/PacketProcessors/Logistics/StationUIInitialSyncRequestProcessor.cs

```diff
@@ -0,0 +1,74 @@
+﻿using NebulaModel.Attributes;
+using NebulaModel.Networking;
+using NebulaModel.Packets.Logistics;
+using NebulaModel.Packets.Processors;
+
+/*
+ * When a client opens a stations UI we sync the complete state of settings and storage.
+ * After that he will receive live updates while the UI is opened.
+ */
+namespace NebulaHost.PacketProcessors.Logistics
+{
+    [RegisterPacketProcessor]
+    public class StationUIInitialSyncRequestProcessor: IPacketProcessor<StationUIInitialSyncRequest>
+    {
+        public void ProcessPacket(StationUIInitialSyncRequest packet, NebulaConnection conn)
+        {
+            StationComponent[] gStationPool = GameMain.data.galacticTransport.stationPool;
+            if(packet.stationGId < gStationPool.Length)
+            {
+                StationComponent stationComponent = null;
+                if(packet.planetId == 0)
+                {
+                    stationComponent = gStationPool[packet.stationGId];
+                }
+                else
+                {
+                    PlanetData pData = GameMain.galaxy.PlanetById(packet.planetId);
+                    // GId is the id in this case as we look at a PLS
+                    if(pData?.factory?.transport != null && pData.factory.transport.stationPool.Length > packet.stationGId)
+                    {
+                        stationComponent = pData.factory.transport.stationPool[packet.stationGId];
+                    }
+                }
+                if(stationComponent != null)
+                {
+                    StationStore[] storage = stationComponent.storage;
+                    int[] itemId = new int[storage.Length];
+                    int[] itemCountMax = new int[storage.Length];
+                    int[] itemCount = new int[storage.Length];
+                    int[] localLogic = new int[storage.Length];
+                    int[] remoteLogic = new int[storage.Length];
+                    int[] remoteOrder = new int[storage.Length];
```

Comment:
I think there is space for future improvement, to have these arrays pre-allocated and just zero it here. Or maybe it does not need to be cleaned, since all values are being replaced right? So maybe just reusing the same arrays, so we would prevent allocation on every packet.